assets/javascripts/discourse/models/filter-rule.js.es6

```diff
@@ -1,17 +1,22 @@
 import RestModel from 'discourse/models/rest';
 
 export default RestModel.extend({
-  category_id: -1,
+  category_id: '',
   channel: '',
   filter: null,
 
   category: function() {
     var id = this.get('category_id');
 
-    if (id === "0")
-      return Discourse.Category.create({ name: 'All Categories', id: 0 });
-    else {
-      return Discourse.Category.findById(id) || { id: id, name: 'Deleted Category' };
+    switch (id) {
+      case '*':
+        return Discourse.Category.create({ name: I18n.t('slack.choose.all_categories'), id: '*' });
+        break;
+      case '':
```

Comment:
Actually "All categories" means all topics here. Because all topics going to be under any one category. It will send slack message for all new topic creations.

For example, if someone created filter like `{channel: "#team", category: "All categories", tags: ["dev", "bug"], filter: "follow"}`. Tags will have no effect here. I mean already Discourse will send a slack message to channel `#team` for all topic creations. So no need to mention the tags here.

In this case it should be  `{channel: "#team", category: nil, tags: ["dev", "bug"], filter: "follow"}`. Then only it can send message to only topics which are associated with mentioned tags.